src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -546,7 +546,11 @@ internal void VerifyWrite(int originalDepth, Utf8JsonWriter writer)
         /// <remarks>Method should be overridden in custom converters of types used in deserialized dictionary keys.</remarks>
         public virtual T ReadAsPropertyName(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            if (!IsInternalConverter && options.TryGetDefaultSimpleConverter(TypeToConvert, out JsonConverter? defaultConverter))
+            if (!IsInternalConverter &&
+                options.SerializerContext is null && // For consistency do not return any default converters for
```

Comment:
this logic somewhat feels wrong here (even though seems you're preserving behavior) - I think we should be using regular converter path here

Reply:
Yes, it's intentionally preserving backward compatibility with .NET 5